extensions/amp-story/1.0/amp-story-base-layer.js

```diff
@@ -38,4 +48,66 @@ export class AmpStoryBaseLayer extends AMP.BaseElement {
   buildCallback() {
     this.element.classList.add('i-amphtml-story-layer');
   }
+
+  /**
+   * Returns true if the page should be prerendered (for being an active page or first page)
+   * @protected
+   * @return {boolean}
+   */
+  isPrerenderActivePage() {
+    if (this.isPrerenderActivePage_ != null) {
+      return this.isPrerenderActivePage_;
+    }
+    const hashId = parseQueryString(this.win.location.href)['page'];
+    let selector = 'amp-story-page:first-of-type';
+    if (hashId) {
+      selector += `, amp-story-page#${escapeCssSelectorIdent(hashId)}`;
+    }
+    const selectorNodes = this.win.document.querySelectorAll(selector);
+    this.isPrerenderActivePage_ =
+      selectorNodes[selectorNodes.length - 1] === this.element.parentElement;
+    return this.isPrerenderActivePage_;
+  }
+
+  /** @protected */
+  initializeAspectRatioListeners() {
```

Comment:
Let's make this method more general about aspect ratio things (presets, aspect-ratio, scaling-factor...).
Each layer (grid and page) should override it and call super(). The page-layer will read different attributes (preset) but still run the same logic, it'd be great to plan for it in this PR.